ddtrace/sampling_rule.py

```diff
@@ -129,9 +133,13 @@ def matches(self, span):
         :returns: Whether this span matches or not
         :rtype: :obj:`bool`
         """
-        # Our LFU cache expects a single key, convert the
-        # provided Span into a hashable tuple for the cache
-        return self._matches((span.service, span.name))
+        # if we're just interested in root spans and this isn't one, then return False
+        if self.target_span == "root":
+            if span.parent_id is not None:
```

Comment:
"we propose to implement an option in tracers to ignore upstream sampling decision." 

We can likely implement this feature through this part of the code right, by adding a check for whatever config option we receive, or if it's a per rule option we can check the rule.

Reply:
> "we propose to implement an option in tracers to ignore upstream sampling decision."

for this, we'd need to ignore it in context propagation extraction? if we ignore it there and then can cause normal root span creation to trigger sampling decision.

I am not confident this is the right call, but what came to mind for me.

Reply:
Ah that makes more sense. Forgot where I was in the code!